x-pack/plugins/security_solution/public/resolver/view/clickthrough.test.tsx

```diff
@@ -149,6 +154,31 @@ describe('Resolver, when analyzing a tree that has two related events for the or
         graphErrorElements: 0,
         originNode: 1,
       });
+
+      originBounds = getComputedNodeBoundaries(entityIDs.origin);
+      firstChildBounds = getComputedNodeBoundaries(entityIDs.firstChild);
+      secondChildBounds = getComputedNodeBoundaries(entityIDs.secondChild);
+      edgeStartingCoordinates = getEdgeStartingCoordinates();
+      edgeTerminalCoordinates = getEdgeTerminalCoordinates();
+    });
+
+    it('should have one and only one outgoing edge from the origin node', () => {
+      // This winnows edges to the one(s) that "start" under the origin node
+      const edgesThatStartUnderneathOrigin = edgeStartingCoordinates.filter(
+        coordinateBoundaryFilter(originBounds)
+      );
+      expect(edgesThatStartUnderneathOrigin).toHaveLength(1);
```

Comment:
maybe you could have a function that gets the `style.top`, `style.left`, `style.transformOrigin`, and `style.transform`. Then you could compare that for each edge line to what you expect. That way the behavior of the test is more obvious and failures will have a clear explanation of what failed.

Reply:
The `coordinateBoundaryFilter` is checking the edglelines for the expected top, left, etc. Maybe I could put in a comment to make that a little more clear to the reader? The "expectation" here is that "one and only one edge line will start underneath the origin node".

Reply:
What about having the expect look like this:
```
expect(whatever).toYieldEqualTo({
  top: '123px',
  left: '234px',
  transformOrigin: 'center left',
  transform: 'translate(-50%, -50%) rotateZ(0.40rad)'
})
```

Reply:
Thought about that first and then decided you wouldn't like it because it's brittle / too sensitive to specific view-level implementation. This test could become a nuisance if it trips every time you nudge a node or an edgeline a few pixels. Testing that the "start of the edgeline lies underneath the node" seems to me to strike the right balance between some assertion of correctness and not being coupled too tightly to some specific implementation of the view (e.g. it is _exactly this many px left_). I'm happy to adjust to the more view-specific approach, I was just trying to follow your past guidance of trying to avoid that here.

Reply:
Maybe I don't understand the tests. Let's talk about it in office hours?